### PR TITLE
Add support for starting thread from oldest replies

### PR DESCRIPTION
**Pull request by nuno-vieira**
### 🔗 Issue Links
Resolves https://github.com/GetStream/ios-issues-tracking/issues/351

### 🎯 Goal
Adding support for threads starting from the oldest replies instead of the newest. (Slack-like)

### 📝 Summary
Some extra stuff was implemented in this PR since it was related.
- Add support for threads starting from the oldest replies when `threadRepliesStartFromOldest` is enabled
- Rename `scrollToMostRecentMessage` -> `scrollToBottom`
- Rename `ScrollToLatestButton` -> `ScrollToBottomButton`
- Improved Channel List prefetching speed when loading more channels. (This one was not related but a quick change, one liner)

The renames were done because it was becoming very confusing. We had 3 names for the same thing. "ScrollToBottom", "ScrollToLatestMessage", and "ScrollToMostRecentMessage" 🤯 ... So it made sense to add consistency here, especially since we are introducing now a new "ScrollToTop".

### 🎨 Showcase

https://github.com/GetStream/stream-chat-swift/assets/12814114/527f0a5f-2002-42dc-b260-885b6d9b054f

### 🧪 Manual Testing Notes
- Open Demo App Configuration
- Enable both "shouldMessagesStartAtTheTop" and "threadRepliesStartFromOldest"
- Open a channel
- When opening a Thread, the replies should start from the oldest replies

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should be manually QAed
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift [hunk 1]
```diff
@@ -151,6 +151,7 @@ open class ChatChannelListVC: _ViewController,
             self?.channelListErrorView.hide()
         }
 
+        viewPaginationHandler.bottomThreshold = 800
```
[the remark this answers is not in the document]
Comment (nuno-vieira) [reply Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift:154]: It is in the Changelog as well as in the PR Description. **"Improve Channel List prefetching when loading more channels"**

File: Sources/StreamChatUI/ChatThread/ChatThreadVC.swift [hunk 2]
```diff
@@ -128,13 +134,22 @@ open class ChatThreadVC: _ViewController,
...
 
-                    self.jumpToMessage(id: initialReplyId)
+                    self?.jumpToMessage(id: initialReplyId)
+                }
+                return
+            }
+
+            // When we tap on the parent message and start from oldest replies is enabled
+            if self.shouldStartFromOldestReplies, let parentMessage = self.messageController.message {
+                self.messageController.loadPageAroundReplyId(parentMessage.id) { [weak self] _ in
```
[the remark this answers is not in the document]
Comment (nuno-vieira) [reply Sources/StreamChatUI/ChatThread/ChatThreadVC.swift:151]: It is how the backend works right now. If we pass the parent message id when loading a mid-page, it loads the oldest page of replies.

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (polqf) [reply Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift:154]: I've seen that 😂  I was asking about the reason for it. Why this is now changed here and not in `ScrollViewPaginationHandler`? Not needed for the ChannelVC?

**Review by polqf — APPROVED** [review]

**Review by testableapple — APPROVED** [review]
LGTM 🚢

**Pull request merged**